bridge-adapters/kinetic-bridgehub-adapter-azure-rest/src/main/java/com/kineticdata/bridgehub/adapter/azure/rest/AzureRestApiHelper.java

```diff
@@ -154,9 +175,16 @@ private JSONObject parseResponse(String output) throws BridgeError{
             throw new BridgeError("An unexpected error has occured " + e);
         }
         
-        if(jsonResponse.get("error") != null) {
-            JSONObject error = (JSONObject)jsonResponse.get("error");
-            throw new BridgeError("Received error: " 
+        // Log the error if it is Authentication because we will get a new
+        // auth token
+        JSONObject error = (JSONObject)jsonResponse.get("error");
+        if(error != null 
+            && error.get("code").toString().equals("InvalidAuthenticationToken")) {
+            
+            LOGGER.error("Received error: "  + error.get("code").toString() 
+                + ", Description: " + error.get("message").toString());
+        } else if (error != null) {
```

Comment:
This `else if (error != null)` will never be run because the `if` above checks the same thing.

This entire method is very buggy with potential class cast exceptions.
For example: line 171 could throw an error if the output isn't a JsonArray, but that won't be caught. I would just handle the errrors in the `handleFailedResponse()` method, and rewrite this entire method as:

```
private JSONObject parseResponse(String output) throws BridgeError{
    try {
        // Parse the response string into a JSONObject
        return (JSONObject)JSONValue.parse(output);
    } catch (ClassCastException e) {
        throw new BridgeError("Failed to parse the response as JSON", e);
    } catch (Exception e) {
        throw new BridgeError("An unexpected error has occurred trying to parse the response", e);
    }
}
```